File: src/main/java/no/fint/provider/adapter/FintAdapterProps.java
package no.fint.provider.adapter;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Getter
@Component
public class FintAdapterProps {

    @Value("${fint.adapter.organizations}")
    private String[] organizations;

    @Value("${fint.adapter.sse-expiration:1200000}")
    private int expiration;

}

File: src/main/kotlin/no/fint/provider/adapter/FintAdapterProps.kt
package no.fint.provider.adapter

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class FintAdapterProps(
    @param:Value($$"${fint.adapter.organizations}")
    val organizations: Array<String>,
    @param:Value($$"${fint.adapter.sse-expiration:1200000}")
    val expiration: Int,
    @param:Value($$"${fint.adapter.reject-unknown-events:true}")
    val rejectUnknownEvents: Boolean
)
